src/textual/await_complete.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from asyncio import Future, gather, wait
+from typing import Any, Coroutine, Generator, Generic, TypeVar
+
+ReturnType = TypeVar("ReturnType")
+
+
+class AwaitComplete(Generic[ReturnType]):
+    """An 'optionally-awaitable' object."""
+
+    _instances: set["AwaitComplete"] = set()
+    """Track all active instances of AwaitComplete."""
+
+    def __init__(self, *coroutine: Coroutine) -> None:
```

Comment:
Could you type the return type of `Coroutine` here. I think that it will allow the checker to infer the return type of the awaitable.